x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLConfigurationSettings.java

```diff
@@ -263,14 +256,10 @@ private SSLConfigurationSettings(String prefix, boolean acceptNonSecurePasswords
             truststoreType,
             trustRestrictionsPath,
             trustRestrictionsX509Fields,
-            caPaths
+            caPaths,
+            clientAuth,
+            verificationMode
         );
-        switch (intendedUse) {
-            case CLIENT -> enabled.add(verificationMode);
-            case SERVER -> enabled.add(clientAuth);
-            case BOTH -> enabled.addAll(List.of(verificationMode, clientAuth));
-            default -> throw new IllegalArgumentException("invalid intended use [" + intendedUse + "]");
-        }
```

Comment:
This essentially reverts the [previous change](https://github.com/elastic/elasticsearch/pull/93614/files#diff-8fd9b5704d17a943fb3d8b6c758a6e89220d7c937452f7978c0db55b252f979aR272-R277) when spliting client and server SSL settings.

In hindsight, the previous change was a premature optimization trying to expose only relevant SSL settings depending on the client/server mode. It is true that `client_authentication` is not useful on the client side. But `verification_mode` is still useful on the server side. What we really want for `verfication_mode` is to somehow tell end-users that "setting it to `full` is equivalent to `certificate` on the server side because server cannot verify client's hostname". But instead, the previous change disabled it entirely which was wrong. 

Given the complexity of generating such message (requires passing the intended usage around) and the fact that all other existing SSL settings, e.g. HTTP, have no such treatment. I decided to take an easy fix to just let remote cluster SSL behaves like existing SSL settings.

Reply:
I'm good with the simpler approach for `verification_mode` -- we can add "setting it to `full` is equivalent to `certificate` on the server side because server cannot verify client's hostname" in the docs (where it makes sense)

Non-blocking sanity check: since `client_authentication` doesn't make sense for client mode, we could still leave the existing validation in place for that setting, correct? Is the reason we're removing it that the departure from other SSL settings behavior is less justified? I'm not opposed just wondering if this might still be useful to keep around.

Reply:
We synced in a separate channel and agreed to keep remote cluster SSL settings consistent with all other existing SSL settings.